### DISCUSSION TITLE
Why `a` do selection but `i` don't?

### DISCUSSION THREAD

**Discussion by eailfly** [helix-editor/helix#2924] [Q&A] [answered]
I see there's a issue #2673 talk about this, but I'm still confused
From my thought, `i` and `a` should behavior just same except different direction, if we say the `a` extends the selection and useful for multiple selections, isn't that `i` should be same too? after all, people don't just 'append' selection, they insert before selection sometimes too.

**Comment by the-mikedavis** [comment, marked as answer]
Currently a cursor must be at either the anchor (beginning) or head (end) of a selection (or both if it's a 1-width selection). If `i` extended the selection, the cursor would end up within the selection which is not allowed. If `a` retained the selection but did not extend it, the cursor would be detached from the selection. You could collapse the selection to the head with `;` before hitting `li` to append without extending or retaining the selection. That behavior would be a worse default though in multi-selection editing scenarios.

Consider this contrived example:

```json
[
  {"data": 87},
  {"data": 88},
  {"data": 89}
]
```

If you wanted to add `.0` to the ends of those digits, you could use

```
%s\d+a.0
```

then if you wanted to wrap those numbers in double quotes you could use `ms"` in normal mode (surrounds the selection with double quotes). If the selection was collapsed by `a`, you'd need to go re-select the digits. And in cases where you want to collapse the selection, you can use `;`.

**Reply by eailfly** [reply]
That make sense, thanks sir

**Reply by sibouras** [reply]
ah i see how this is useful thanks, but this also can be solved by `gv`  https://github.com/helix-editor/helix/issues/6695 and keep `a` more intuitive
